### ISSUE TITLE
unexpected EOF

### ISSUE THREAD

**Issue by kharabela** [gabriel-vasile/mimetype#450]
go install "github.com/gabriel-vasile/mimetype@latest" 
downloading github.com/gabriel-vasile/mimetype v1.4.3
go: github.com/gabriel-vasile/mimetype@latest: github.com/gabriel-vasile/mimetype@v1.4.3: read "https:/proxy.golang.org/@v/v1.4.3.zip": unexpected EOF

**Comment by ddennedy-gpsw** [comment]
I experience this too, when I try to get it with wget I get more information:
```txt
% wget https://proxy.golang.org/github.com/gabriel-vasile/mimetype/@v/v1.4.5.zip
--2024-09-20 14:24:25--  https://proxy.golang.org/github.com/gabriel-vasile/mimetype/@v/v1.4.5.zip
Resolving proxy.golang.org (proxy.golang.org)... 142.250.217.145
Connecting to proxy.golang.org (proxy.golang.org)|142.250.217.145|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://storage.googleapis.com/proxy-golang-org-prod/0fa05c41132a5513-github.com:gabriel-vasile:mimetype-v1.4.5.zip?Expires=1726950903&GoogleAccessId=gcs-urlsigner-prod%40golang-modproxy.iam.gserviceaccount.com&Signature=P8%2FSpFsBX3SHeVLWIga3VhcmMtj58QbZeSUsTIJHQ2FZ2r%2BDnk%2FUVjTB5w01X0YXyofr83QtBwLpqMDVLA1eIbZskZJ74EzdFFN%2BuX%2FhzoQ2kIY5cX587t%2FqVdW9R5oNAZLTYu2BFWH9mI%2FOa7kiSEAe8rzk4fV%2BAqnFYXe%2B7mm4BnO8G2juTcXLIZQdj8K1ex4WCARiJhidvUjs7g0OK5ZmXIHol2uJ%2FMZgUUQMhycDSBFKt7Ebe1ewUSpSZFdR20K48c3OuaGlYFvJTfP9Vw2KKSnVYRYYlz8iZCvh2LSEvkVWTRSZvYsEF14Ba1QyyvmTIu3UhQtGEBJNmEOjYA%3D%3D [following]
--2024-09-20 14:24:25--  https://storage.googleapis.com/proxy-golang-org-prod/0fa05c41132a5513-github.com:gabriel-vasile:mimetype-v1.4.5.zip?Expires=1726950903&GoogleAccessId=gcs-urlsigner-prod%40golang-modproxy.iam.gserviceaccount.com&Signature=P8%2FSpFsBX3SHeVLWIga3VhcmMtj58QbZeSUsTIJHQ2FZ2r%2BDnk%2FUVjTB5w01X0YXyofr83QtBwLpqMDVLA1eIbZskZJ74EzdFFN%2BuX%2FhzoQ2kIY5cX587t%2FqVdW9R5oNAZLTYu2BFWH9mI%2FOa7kiSEAe8rzk4fV%2BAqnFYXe%2B7mm4BnO8G2juTcXLIZQdj8K1ex4WCARiJhidvUjs7g0OK5ZmXIHol2uJ%2FMZgUUQMhycDSBFKt7Ebe1ewUSpSZFdR20K48c3OuaGlYFvJTfP9Vw2KKSnVYRYYlz8iZCvh2LSEvkVWTRSZvYsEF14Ba1QyyvmTIu3UhQtGEBJNmEOjYA%3D%3D
Resolving storage.googleapis.com (storage.googleapis.com)... 142.251.40.59, 142.250.68.27, 142.250.72.251, ...
Connecting to storage.googleapis.com (storage.googleapis.com)|142.251.40.59|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 24057350 (23M) [application/zip]
Saving to: ‘v1.4.5.zip.1’

v1.4.5.zip.1                                99%[=====================================================================================> ]  22.94M  18.0MB/s    in 1.3s    

2024-09-20 14:24:27 (18.0 MB/s) - Read error at byte 24054833/24057350 (Connection reset by peer). Retrying.

--2024-09-20 14:24:28--  (try: 2)  https://storage.googleapis.com/proxy-golang-org-prod/0fa05c41132a5513-github.com:gabriel-vasile:mimetype-v1.4.5.zip?Expires=1726950903&GoogleAccessId=gcs-urlsigner-prod%40golang-modproxy.iam.gserviceaccount.com&Signature=P8%2FSpFsBX3SHeVLWIga3VhcmMtj58QbZeSUsTIJHQ2FZ2r%2BDnk%2FUVjTB5w01X0YXyofr83QtBwLpqMDVLA1eIbZskZJ74EzdFFN%2BuX%2FhzoQ2kIY5cX587t%2FqVdW9R5oNAZLTYu2BFWH9mI%2FOa7kiSEAe8rzk4fV%2BAqnFYXe%2B7mm4BnO8G2juTcXLIZQdj8K1ex4WCARiJhidvUjs7g0OK5ZmXIHol2uJ%2FMZgUUQMhycDSBFKt7Ebe1ewUSpSZFdR20K48c3OuaGlYFvJTfP9Vw2KKSnVYRYYlz8iZCvh2LSEvkVWTRSZvYsEF14Ba1QyyvmTIu3UhQtGEBJNmEOjYA%3D%3D
Connecting to storage.googleapis.com (storage.googleapis.com)|142.251.40.59|:443... connected.
HTTP request sent, awaiting response... 403 Forbidden
2024-09-20 14:24:28 ERROR 403: Forbidden.
```

**Comment by gabriel-vasile** [comment]
I was never able to reproduce this. Maybe it was related to big download size (due to the testdata directory in this repo).

If that was the case, it should be fixed after #603. Feel free to re-open this issue if still not fixed.

Maybe related: #425
